website/server/models/message.js

```diff
@@ -45,6 +45,7 @@ const inboxSchema = new mongoose.Schema({
   // we store two copies of each inbox messages:
   // one for the sender and one for the receiver
   ownerId: { $type: String, ref: 'User' },
+  read: { $type: Boolean, default: false },
```

Comment:
My main concern with this one is that we'll have to run an huge migration to mark as read all existing messages and we actually won't know if they've been read or not